src/components/Dashboard/index.jsx

```diff
@@ -1,38 +1,79 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import './dashboard.scss';
 import Card from './Card';
+import { fetchRealTimeReport } from '../../redux/actions/realTimeReport';
+import listenToSocketEvent from '../../realTime';
+import ActivityFeed from './ActivityFeed';
 
 class Dashboard extends React.Component {
-  /*
-  Use case Example for dashboard card
-  component is a function that returns a component
-  Suppose you are working on Upselling Partners task
-  and you have created a class component named Upselling.
-  You will create a function renderUpselling that returns
-  your Upselling jsx component and pass it as props.
+  state = {
+    reportData: [],
+  };
+
+  componentDidMount = () => {
+    // triggers event to fetch data from the API
+    this.connectToSocket('newAutomation');
+  };
+
+  connectToSocket = (event) => {
+    listenToSocketEvent(event, (data) => {
+      this.handleDataFetch(data);
+    });
+  };
+
+  /**
+   * @description handles fetching of data and adds to the state
+   * @param {object} data coming in from the socket
+   * @returns {Void}
+   */
+  handleDataFetch = (data) => {
+    const { fetchUpdates } = this.props;
+    fetchUpdates(data);
+    const { reportData } = this.state;
+    const newData = this.maintainArraySize(data, reportData);
+    this.setState({ reportData: newData });
+  };
+
+  /**
+   * @description keeps data in an array to the amount set
+   * @param {object} data to be put inside an array
+   * @param {array} arr
+   * @returns {*[]} an array with new data
    */
-  renderUpselling = () => (
-    <div>
-        upselling component
-    </div>
-  );
+  maintainArraySize = (data, arr) => {
+    const newArr = [data, ...arr];
+    if (arr.length > process.env.REACT_APP_ACTIVITY_FEED_NUMBER) {
+      newArr.pop();
```

Comment:
Quick question, does this mean that the card will not display all the information fetched but will display a section or number of the information dependent on the value set in `.env` file

Reply:
Short Answer: The card will only display a certain number of items
Long Answer: I had a sync with Arnold and he suggested that instead of rendering all the feeds that come in, we should have a fixed number that can be adjusted.

Reply:
Ooh okay thanks.